tracpro/contacts/models.py

```diff
@@ -191,19 +208,20 @@ def get_urn(self):
     def kwargs_from_temba(cls, org, temba_contact):
         """Get data to create a Contact instance from a Temba object."""
         def _get_first(model_class, temba_objects):
-            """Return first obj from this org that matches one of the given uuids."""
+            """Return first obj from this org that matches one of the given uuids, or None."""
             queryset = model_class.get_all(org)
-            tracpro_uuids = queryset.values_list('uuid', flat=True)
-            temba_uuids = [temba_object.uuid for temba_object in temba_objects]
-            uuid = next((uuid for uuid in temba_uuids if uuid in tracpro_uuids), None)
-            return queryset.get(uuid=uuid) if uuid else None
+            temba_uuids = get_uuids(temba_objects)
+            return queryset.filter(uuid__in=temba_uuids).first()
         # Use the first Temba group that matches one of the org's Regions.
         region = _get_first(Region, temba_contact.groups)
         if not region:
-            raise ValueError(
-                "Unable to save contact {c.uuid} ({c.name}) because none of "
-                "their cohorts match an active Panel for this org.".format(
-                    c=temba_contact))
+            raise NoMatchingCohortsWarning(("Unable to save contact {c.uuid} ({c}) because none of "
+                                            "their cohorts ({cohorts}) match an active Panel "
+                                            "({panels}) for this org.").format(
```

Comment:
We talked about this in chat already, but let's make this more readable by using including the contact's name and excluding `{cohorts}` and `{panels}`